### PR TITLE
User support for blank character

### PR DESCRIPTION
**Pull request by TribesTom**
Following this https://community.openproject.org/work_packages/17835
Here is the merge request to change it.

Have a good day.

### CODE REVIEW

**Comment by TeatroIO** [comment]
I've prepared a stage to preview changes. <a href='http://tribestom-release-4-0.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://tribestom-release-4-0.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

File: app/models/user.rb [hunk 1]
```diff
@@ -144,7 +144,7 @@ def self.create_blocked_scope(blocked)
   validates_uniqueness_of :login, :if => Proc.new { |user| !user.login.blank? }, :case_sensitive => false
   validates_uniqueness_of :mail, :allow_blank => true, :case_sensitive => false
   # Login must contain letters, numbers, underscores only
-  validates_format_of :login, :with => /\A[a-z0-9_\-@\.]*\z/i
+  validates_format_of :login, :with => /\A[a-z0-9_\-@\.\s]*\z/i
```
Comment (myabc) [inline app/models/user.rb:148]: @TribesTom I think we can ignore the Hound warnings here, as the original code didn't conform either (style issues have been fixed on `dev`, but not on the `release/4.0` branch).

`\s` doesn't strike me as the right choice here – we don't want to allow tabs or new line characters. AFAICT we only need to allow ` `.

**Comment by TribesTom** [comment]
Hi myabc

I don't know which regex to use to only accept " " character and not new line and tab.

Thanks for your time

**Comment by ulferts** [comment]
@TribesTom, I think `/\A[a-z0-9_\-@\. ]*\z/i` (please notice the blank at the end of the group) should do what you intend.

BTW, does anybody know why the \* is at the end of the group as we do not allow an empty login? Is it only to prevent to have two error messages for an empty login field?

**Comment by ulferts** [comment]
@TribesTom could you please take another look at the PR and assess whether the change in the regexp from \s to " " would be fine?

The PR is getting old now and I would like to merge it in soon.

**Comment by TribesTom** [comment]
Sorry for late response.

I'll test and I ll tell you if it s ok.

2015-02-23 13:33 GMT+01:00 ulferts notifications@github.com:

**Comment by ulferts** [comment]
@TribesTom thanks!

**Comment by TribesTom** [comment]
Hi Back.

It seems it's ok with /\A[a-z0-9_-@. ]*\z/i  

Thank you.

**Comment by NobodysNightmare** [comment]
Looks like this PR totally got lost unnoticed.

@myabc If I see this correctly, it should be fine if we replaced the `\s` with a single space, right?

Apparently @TribesTom did forget to change the actual code here, but as it is a one-liner I would go ahead and provide the neccessary PR for that.

**Comment by myabc** [comment]
@NobodysNightmare this needs a test IMO

**Comment by floriank** [comment]
@NobodysNightmare Please go ahead and add the test, Use the commits in this PR as a base, so that @TribesTom's contributions do not go unnoticed :+1:

**Comment by NobodysNightmare** [comment]
Closing this PR in favor of the new one.

Contribution by @TribesTom stays visible ;-)

**Pull request closed without merging**